src/js/components/List/__tests__/List-test.js

```diff
@@ -527,4 +527,36 @@ describe('List onOrder', () => {
     expect(onOrder).toHaveBeenCalled();
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  test('itemKey', () => {
+    const { container } = render(
+      <Grommet>
+        <List
+          data={[
+            { a: 'one', b: 1 },
+            { a: 'two', b: 2 },
+          ]}
+          primaryKey="a"
+          itemKey="b"
+        />
+      </Grommet>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
```

Comment:
The snapshots don't show the key. Can we change the tests to just check what key got applied to the list item

Reply:
Correct me if I am wrong, after researching, I don't think there is a good way to test which key got applied to the list.
https://stackoverflow.com/questions/69758600/how-can-i-assert-which-value-has-been-passed-to-a-li-key-attribute-in-react-wi

However, if we can specify `id={key}` prop to the `StyledItem`, then it might be possible to capture the value of key in the snapshots. Please let me know your thoughts.

Reply:
I took some time to think about this more and I agree that there isn't a good way to test this. While we could do something to be able to access and test the key prop I am thinking that testing for specific prop values isn't something we want to be doing in the first place. I found this great article on why this type of testing isn't a good practice: https://kentcdodds.com/blog/testing-implementation-details.

Anyway sorry for the back and forth on this. I think we can remove the tests and then this PR will be ready to go!